<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane L Hogle">
<meta name="dcterms.date" content="2025-07-20">
<meta name="keywords" content="Microbiology, Ecology, Evolution">

<title>RNAseq ata analysis for the project ‘bacteria-nematode interactions’ – RNAseq bacteria-nematode interactions</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">
      RNAseq bacteria-nematode interactions
      </li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">RNAseq bacteria-nematode interactions</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/01_read_format_featureCounts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Formatting RNAseq counts</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./R/02_dge_function.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Analysis and plotting</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#overview" id="toc-overview" class="nav-link active" data-scroll-target="#overview"><span class="header-section-number">1</span> Overview</a>
  <ul class="collapse">
  <li><a href="#format-122-species-syncom" id="toc-format-122-species-syncom" class="nav-link" data-scroll-target="#format-122-species-syncom"><span class="header-section-number">1.1</span> Format 122 species SynCom</a></li>
  <li><a href="#preparing-for-read-mapping-and-feature-counting" id="toc-preparing-for-read-mapping-and-feature-counting" class="nav-link" data-scroll-target="#preparing-for-read-mapping-and-feature-counting"><span class="header-section-number">1.2</span> Preparing for read mapping and feature counting</a></li>
  <li><a href="#quality-control-rna-seq-data" id="toc-quality-control-rna-seq-data" class="nav-link" data-scroll-target="#quality-control-rna-seq-data"><span class="header-section-number">1.3</span> Quality control RNA-seq data</a></li>
  <li><a href="#map-rna-seq-data" id="toc-map-rna-seq-data" class="nav-link" data-scroll-target="#map-rna-seq-data"><span class="header-section-number">1.4</span> Map RNA-seq data</a></li>
  <li><a href="#process-data-for-differential-expression" id="toc-process-data-for-differential-expression" class="nav-link" data-scroll-target="#process-data-for-differential-expression"><span class="header-section-number">1.5</span> Process data for differential expression</a></li>
  <li><a href="#compare-samples-via-ordination" id="toc-compare-samples-via-ordination" class="nav-link" data-scroll-target="#compare-samples-via-ordination"><span class="header-section-number">1.6</span> Compare Samples via Ordination</a></li>
  <li><a href="#find-differentially-expressed-genes" id="toc-find-differentially-expressed-genes" class="nav-link" data-scroll-target="#find-differentially-expressed-genes"><span class="header-section-number">1.7</span> Find differentially expressed genes</a></li>
  <li><a href="#availability" id="toc-availability" class="nav-link" data-scroll-target="#availability"><span class="header-section-number">1.8</span> Availability</a></li>
  <li><a href="#reproducibility" id="toc-reproducibility" class="nav-link" data-scroll-target="#reproducibility"><span class="header-section-number">1.9</span> Reproducibility</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">RNAseq ata analysis for the project ‘bacteria-nematode interactions’</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Shane L Hogle <a href="https://orcid.org/0000-0003-2595-3863" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://www.utu.fi/en">
            University of Turku
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 20, 2025</p>
    </div>
  </div>
  
    
  </div>
  

<div>
  <div class="keywords">
    <div class="block-title">Keywords</div>
    <p>Microbiology, Ecology, Evolution</p>
  </div>
</div>

</header>


<section id="overview" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Overview</h1>
<p>This work is done in collaboration with Dr.&nbsp;Liu Ting, Dr.&nbsp;Li Gen, and Prof.&nbsp;Zhong Wei at Nanjing Agriculturatl University. Here we quality control the RNAseq data from a metatranscriptomics experiment, map to the Nanjing 122 bacterial species SynCom, and count reads mapping to genes.</p>
<p>In subsequent steps we process the output in R so that we can provide it to our collaborators.</p>
<section id="format-122-species-syncom" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="format-122-species-syncom"><span class="header-section-number">1.1</span> Format 122 species SynCom</h2>
<p>Note this code is not meant to be executed in this project directory but is meant to provide a record of the commands used to format and process the necessary data files from the SynCom assemblies.</p>
<ol type="1">
<li>Formatting assemblies</li>
</ol>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># Get the ids of the genome assemblies</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">mkdir</span> assemblies</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="bu">cd</span> assemblies_orig</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="bu">printf</span> <span class="st">'%s\n'</span> <span class="pp">*</span>.fna <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.fna//g'</span> <span class="op">&gt;</span> ids.txt</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">mv</span> ids.txt ..</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="bu">cd</span> ..</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>mkdir: cannot create directory ‘assemblies’: File exists
bash: line 3: cd: assemblies_orig: No such file or directory</code></pre>
</div>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># Here we rename all chromosomes so that they have a single name, appending an</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co"># integer if there are multiple sequences per genome</span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="fu">cat</span> ids.txt <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">ID</span><span class="kw">;</span> <span class="cf">do</span> <span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">".*"</span> <span class="at">-r</span> <span class="va">$ID</span> assemblies_orig/<span class="va">$ID</span>.fna <span class="kw">|</span> <span class="ex">seqkit</span> rename <span class="at">-n</span> <span class="kw">|</span> <span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"\s.*$"</span> <span class="at">-r</span> <span class="st">""</span> <span class="kw">|</span> <span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"_"</span> <span class="at">-r</span> <span class="st">"."</span> <span class="op">&gt;</span> assemblies/<span class="va">$ID</span>.fna<span class="kw">;</span> <span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>cat: ids.txt: No such file or directory</code></pre>
</div>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource bash number-lines code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># Now we find assemblies with multiple chromosomes for some later manual tweaks.</span></span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="co"># This is needed to have a consistent naming scheme</span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"&gt;"</span> assemblies/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/\:/ /g'</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-nrk2,2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>grep: assemblies/*: No such file or directory</code></pre>
</div>
</div>
<ol start="2" type="1">
<li>Prokka</li>
</ol>
<p>Run prokka to call genes on the 122 assemblies and annotate them with a consistent format - see <code>sh/prokka/prokka.sh</code></p>
<ol start="3" type="1">
<li>Eggnoggmapper</li>
</ol>
<p>Run eggnoggmapper to functionally annotate genes - see <code>sh/eggnoggmapper/eggnoggmapper.sh</code></p>
<ol start="4" type="1">
<li>Formatting for rRNA filtering (these steps performed on computer cluster)</li>
</ol>
<p>Get rRNAs from prokka which will be used later in the quality control steps to filter out reads mapping to bacterial rRNAs</p>
<pre><code>cd prokka
grep "ribosomal RNA" */*.tsv | sed 's/.tsv:/.ffn\t/g' | cut -f1,2 | while read FILE RNA; do seqkit grep -n -r -p "${RNA}" ${FILE}; done &gt; nanjingSynCom122_rrnas.fasta
seqkit rmdup -s nanjingSynCom122_rrnas.fasta</code></pre>
</section>
<section id="preparing-for-read-mapping-and-feature-counting" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="preparing-for-read-mapping-and-feature-counting"><span class="header-section-number">1.2</span> Preparing for read mapping and feature counting</h2>
<ol type="1">
<li>Make SAF file from GFF file to be used with featureCounts</li>
</ol>
<pre><code>cd rnaseq/map_count/saf
cat ../../../nanjingSynCom122/ids.txt | while read ID; do sed -n '/&gt;/q;p' ../../../nanjingSynCom122/prokka/${ID}/${ID}.gff | grep -v "^#" | sed 's/;.*$//g' | sed 's/ID=//g' | bioawk -t '{print $9, $1, $4, $5, $7}' | grep -v "^note"; done &gt; nanjingSynCom122_combined.saf</code></pre>
<p>The resulting file looks like this</p>
<pre><code>head saf/nanjingSynCom122_combined.saf</code></pre>
<pre><code>GeneID  Chr Start   End Strand
1aci1_00001 1aci1   178 1365    +
1aci1_00002 1aci1   1498    2616    +
1aci1_00003 1aci1   2754    5378    +
1aci1_00004 1aci1   5603    5920    -
1aci1_00005 1aci1   6257    6868    -
1aci1_00006 1aci1   7122    7913    +
1aci1_00007 1aci1   7967    8389    +
1aci1_00008 1aci1   8463    8807    -
1aci1_00009 1aci1   8986    9246    +</code></pre>
<p>Concatenate all assemblies into single file for mapping with bwa. Create the necessary indexes</p>
<pre><code>cd rnaseq/map_count/refs
cat ../../../nanjingSynCom122/prokka/*/*.fna &gt; nanjingSynCom122_combined.fna
samtools faidx nanjingSynCom122_combined.fna
bwa index nanjingSynCom122_combined.fna</code></pre>
</section>
<section id="quality-control-rna-seq-data" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="quality-control-rna-seq-data"><span class="header-section-number">1.3</span> Quality control RNA-seq data</h2>
<p>Here we do standard filtering and trimming of reads.Bacterial community total cDNA read pairs are processed using BBDuk (version 39.23 (https://sourceforge.net/projects/bbmap/) to remove contaminants, trim reads that contained adapter sequence, right quality trim reads where quality drops below Q10, and exclude reads with more than 2 Ns. BBMap (version 39.23) was then used map cDNA read pairs to a database of bactieral 16S, 5S and 18S rRNAs from the 122 bacterial species SynCom based on exact matches.</p>
<p>The steps to reproduce this are available in <code>sh/qualitycontrol/</code></p>
</section>
<section id="map-rna-seq-data" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="map-rna-seq-data"><span class="header-section-number">1.4</span> Map RNA-seq data</h2>
<p>Here we map the quality controlled read pairs to a concatenated reference of the 122 bacterial SynCom genomes using BWA. The resulting BAM files are sorted and then the program featureCounts v2.1.1 is used to count the number of reads mapping to genes in the assemblies.</p>
<p>The steps to reproduce this are available in <code>sh/map_count/</code></p>
</section>
<section id="process-data-for-differential-expression" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="process-data-for-differential-expression"><span class="header-section-number">1.5</span> Process data for differential expression</h2>
<ol start="6" type="1">
<li>Load big matrix into R and use gene identifiers to partition into n &lt;= 122 species-specific sub-matrices</li>
</ol>
</section>
<section id="compare-samples-via-ordination" class="level2" data-number="1.6">
<h2 data-number="1.6" class="anchored" data-anchor-id="compare-samples-via-ordination"><span class="header-section-number">1.6</span> Compare Samples via Ordination</h2>
<p>https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02337-8</p>
<ol start="7" type="1">
<li>Calculate TMM for each species sub-matrix separately.</li>
</ol>
<ul>
<li>https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25</li>
<li>https://www.biostars.org/p/9475236/</li>
<li>https://www.biostars.org/p/306010/</li>
<li>https://www.biostars.org/p/419018/#419177</li>
<li>https://davetang.github.io/muse/edger.html</li>
</ul>
<ol start="8" type="1">
<li>Combine TMM matrices for all species into a single matrix and use in a clustering analysis like tSNE (reproduces Fig. 3d)</li>
</ol>
<blockquote class="blockquote">
<p>For the comparison of transcript abundance across multiple samples, transcript counts need to be normalized in each organism by individual transcript lengths and total library size. Due to the differences in transcript abundances between the major and minor organisms, counts for each organism should be normalized independently using transcript per million (TPM) calculations [73]. TPM values are calculated by dividing all read counts by the length of each gene in kilobases to obtain a reads per kilobase (RPK) value for each gene [92]. The RPK value for each gene is then divided by the sum of RPK values divided by 1,000,000. While RPKM and FPKM calculations are also used for normalization, the sum of the RPKM and FPKM values differ between samples with differing numbers of reads, which can result in disproportionate comparisons [92].</p>
</blockquote>
</section>
<section id="find-differentially-expressed-genes" class="level2" data-number="1.7">
<h2 data-number="1.7" class="anchored" data-anchor-id="find-differentially-expressed-genes"><span class="header-section-number">1.7</span> Find differentially expressed genes</h2>
<ol start="9" type="1">
<li>Run a separate edgeR analysis for each species sub-matrix</li>
<li>Concatenate results of significantly differentially expressed genes</li>
<li>Send to collaborators</li>
</ol>
</section>
<section id="availability" class="level2" data-number="1.8">
<h2 data-number="1.8" class="anchored" data-anchor-id="availability"><span class="header-section-number">1.8</span> Availability</h2>
<p>Data and code in this GitHub repository (<a href="https://github.com/GHUSERNAME/REPOSITORYNAME" class="uri">https://github.com/GHUSERNAME/REPOSITORYNAME</a>) are provided under <a href="https://www.gnu.org/licenses/agpl-3.0.html">GNU AGPL3</a>. The rendered project site is available at <a href="https://GHUSERNAME.github.io/REPOSITORYNAME/" class="uri">https://GHUSERNAME.github.io/REPOSITORYNAME/</a>, which has been produced using <a href="https://quarto.org/">Quarto notebooks</a>. The content on the rendered site is released under the <a href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0.</a> This repository hosts all code and data for this project, including the code necessary to fully recreate the rendered webpage.</p>
<p>An archived release of the code is available from Zenodo: <a href="https://zenodo.org/records/EVENTUAL_ZENODO_RECORD" class="uri">https://zenodo.org/records/EVENTUAL_ZENODO_RECORD</a></p>
<p>Raw sequencing data used in the project is available from NCBI Bioproject [PRJNA00000000](https://www.ncbi.nlm.nih</p>
</section>
<section id="reproducibility" class="level2" data-number="1.9">
<h2 data-number="1.9" class="anchored" data-anchor-id="reproducibility"><span class="header-section-number">1.9</span> Reproducibility</h2>
<p>The project uses <a href="https://rstudio.github.io/renv/index.html"><code>renv</code></a> to create a reproducible environment to execute the code in this project. <a href="https://rstudio.github.io/renv/articles/renv.html#collaboration">See here</a> for a brief overview on collaboration and reproduction of the entire project. To get up and running from an established repository, you could do:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>To initiate <code>renv</code> for a new project:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="co"># initialize</span></span>
<span id="cb13-3"><a href="#cb13-3"></a>renv<span class="sc">::</span><span class="fu">init</span>()</span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="co"># install some new packages</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>renv<span class="sc">::</span><span class="fu">install</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb13-6"><a href="#cb13-6"></a><span class="co"># record those packages in the lockfile</span></span>
<span id="cb13-7"><a href="#cb13-7"></a>renv<span class="sc">::</span><span class="fu">snapshot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


<!-- -->

</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a></div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb14" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb14-1"><a href="#cb14-1"></a><span class="co">---</span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="an">title:</span><span class="co"> "RNAseq ata analysis for the project 'bacteria-nematode interactions'"</span></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb14-4"><a href="#cb14-4"></a><span class="an">author:</span></span>
<span id="cb14-5"><a href="#cb14-5"></a><span class="co">  - name: Shane L Hogle</span></span>
<span id="cb14-6"><a href="#cb14-6"></a><span class="co">    orcid: 0000-0003-2595-3863</span></span>
<span id="cb14-7"><a href="#cb14-7"></a><span class="co">    affiliation: </span></span>
<span id="cb14-8"><a href="#cb14-8"></a><span class="co">      - name: University of Turku</span></span>
<span id="cb14-9"><a href="#cb14-9"></a><span class="co">        city: Turku</span></span>
<span id="cb14-10"><a href="#cb14-10"></a><span class="co">        state: Finland</span></span>
<span id="cb14-11"><a href="#cb14-11"></a><span class="co">        url: https://www.utu.fi/en</span></span>
<span id="cb14-12"><a href="#cb14-12"></a><span class="co">#abstract: &gt; </span></span>
<span id="cb14-13"><a href="#cb14-13"></a><span class="an">keywords:</span></span>
<span id="cb14-14"><a href="#cb14-14"></a><span class="co">  - Microbiology</span></span>
<span id="cb14-15"><a href="#cb14-15"></a><span class="co">  - Ecology</span></span>
<span id="cb14-16"><a href="#cb14-16"></a><span class="co">  - Evolution</span></span>
<span id="cb14-17"><a href="#cb14-17"></a><span class="an">license:</span><span class="co"> "CC BY"</span></span>
<span id="cb14-18"><a href="#cb14-18"></a><span class="an">copyright:</span><span class="co"> </span></span>
<span id="cb14-19"><a href="#cb14-19"></a><span class="co">  holder: Shane L Hogle</span></span>
<span id="cb14-20"><a href="#cb14-20"></a><span class="co">  year: 2025</span></span>
<span id="cb14-21"><a href="#cb14-21"></a><span class="co"># citation: </span></span>
<span id="cb14-22"><a href="#cb14-22"></a><span class="co">#   container-title: BioRxiv</span></span>
<span id="cb14-23"><a href="#cb14-23"></a><span class="co">#   volume: </span></span>
<span id="cb14-24"><a href="#cb14-24"></a><span class="co">#   issue: </span></span>
<span id="cb14-25"><a href="#cb14-25"></a><span class="co">#   doi: </span></span>
<span id="cb14-26"><a href="#cb14-26"></a><span class="an">funding:</span><span class="co"> "The author received no specific funding for this work."</span></span>
<span id="cb14-27"><a href="#cb14-27"></a><span class="co">#bibliography: references.bib</span></span>
<span id="cb14-28"><a href="#cb14-28"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb14-29"><a href="#cb14-29"></a><span class="co">  markdown: </span></span>
<span id="cb14-30"><a href="#cb14-30"></a><span class="co">    wrap: sentence</span></span>
<span id="cb14-31"><a href="#cb14-31"></a><span class="co">---</span></span>
<span id="cb14-32"><a href="#cb14-32"></a></span>
<span id="cb14-33"><a href="#cb14-33"></a><span class="fu"># Overview</span></span>
<span id="cb14-34"><a href="#cb14-34"></a></span>
<span id="cb14-35"><a href="#cb14-35"></a>This work is done in collaboration with Dr. Liu Ting, Dr. Li Gen, and Prof. Zhong Wei at Nanjing Agriculturatl University.</span>
<span id="cb14-36"><a href="#cb14-36"></a>Here we quality control the RNAseq data from a metatranscriptomics experiment, map to the Nanjing 122 bacterial species SynCom, and count reads mapping to genes.</span>
<span id="cb14-37"><a href="#cb14-37"></a></span>
<span id="cb14-38"><a href="#cb14-38"></a>In subsequent steps we process the output in R so that we can provide it to our collaborators.</span>
<span id="cb14-39"><a href="#cb14-39"></a></span>
<span id="cb14-40"><a href="#cb14-40"></a><span class="fu">## Format 122 species SynCom</span></span>
<span id="cb14-41"><a href="#cb14-41"></a></span>
<span id="cb14-42"><a href="#cb14-42"></a>Note this code is not meant to be executed in this project directory but is meant to provide a record of the commands used to format and process the necessary data files from the SynCom assemblies.</span>
<span id="cb14-43"><a href="#cb14-43"></a></span>
<span id="cb14-44"><a href="#cb14-44"></a><span class="ss">1.  </span>Formatting assemblies</span>
<span id="cb14-45"><a href="#cb14-45"></a></span>
<span id="cb14-48"><a href="#cb14-48"></a><span class="in">```{bash}</span></span>
<span id="cb14-49"><a href="#cb14-49"></a><span class="co"># Get the ids of the genome assemblies</span></span>
<span id="cb14-50"><a href="#cb14-50"></a><span class="fu">mkdir</span> assemblies</span>
<span id="cb14-51"><a href="#cb14-51"></a><span class="bu">cd</span> assemblies_orig</span>
<span id="cb14-52"><a href="#cb14-52"></a><span class="bu">printf</span> <span class="st">'%s\n'</span> <span class="pp">*</span>.fna <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.fna//g'</span> <span class="op">&gt;</span> ids.txt</span>
<span id="cb14-53"><a href="#cb14-53"></a><span class="fu">mv</span> ids.txt ..</span>
<span id="cb14-54"><a href="#cb14-54"></a><span class="bu">cd</span> ..</span>
<span id="cb14-55"><a href="#cb14-55"></a><span class="in">```</span></span>
<span id="cb14-56"><a href="#cb14-56"></a></span>
<span id="cb14-59"><a href="#cb14-59"></a><span class="in">```{bash}</span></span>
<span id="cb14-60"><a href="#cb14-60"></a><span class="co"># Here we rename all chromosomes so that they have a single name, appending an</span></span>
<span id="cb14-61"><a href="#cb14-61"></a><span class="co"># integer if there are multiple sequences per genome</span></span>
<span id="cb14-62"><a href="#cb14-62"></a><span class="fu">cat</span> ids.txt <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">ID</span><span class="kw">;</span> <span class="cf">do</span> <span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">".*"</span> <span class="at">-r</span> <span class="va">$ID</span> assemblies_orig/<span class="va">$ID</span>.fna <span class="kw">|</span> <span class="ex">seqkit</span> rename <span class="at">-n</span> <span class="kw">|</span> <span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"\s.*$"</span> <span class="at">-r</span> <span class="st">""</span> <span class="kw">|</span> <span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"_"</span> <span class="at">-r</span> <span class="st">"."</span> <span class="op">&gt;</span> assemblies/<span class="va">$ID</span>.fna<span class="kw">;</span> <span class="cf">done</span></span>
<span id="cb14-63"><a href="#cb14-63"></a><span class="in">```</span></span>
<span id="cb14-64"><a href="#cb14-64"></a></span>
<span id="cb14-67"><a href="#cb14-67"></a><span class="in">```{bash}</span></span>
<span id="cb14-68"><a href="#cb14-68"></a><span class="co"># Now we find assemblies with multiple chromosomes for some later manual tweaks.</span></span>
<span id="cb14-69"><a href="#cb14-69"></a><span class="co"># This is needed to have a consistent naming scheme</span></span>
<span id="cb14-70"><a href="#cb14-70"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="st">"&gt;"</span> assemblies/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/\:/ /g'</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-nrk2,2</span></span>
<span id="cb14-71"><a href="#cb14-71"></a><span class="in">```</span></span>
<span id="cb14-72"><a href="#cb14-72"></a></span>
<span id="cb14-73"><a href="#cb14-73"></a><span class="ss">2.  </span>Prokka</span>
<span id="cb14-74"><a href="#cb14-74"></a></span>
<span id="cb14-75"><a href="#cb14-75"></a>Run prokka to call genes on the 122 assemblies and annotate them with a consistent format - see <span class="in">`sh/prokka/prokka.sh`</span></span>
<span id="cb14-76"><a href="#cb14-76"></a></span>
<span id="cb14-77"><a href="#cb14-77"></a><span class="ss">3.  </span>Eggnoggmapper</span>
<span id="cb14-78"><a href="#cb14-78"></a></span>
<span id="cb14-79"><a href="#cb14-79"></a>Run eggnoggmapper to functionally annotate genes - see <span class="in">`sh/eggnoggmapper/eggnoggmapper.sh`</span></span>
<span id="cb14-80"><a href="#cb14-80"></a></span>
<span id="cb14-81"><a href="#cb14-81"></a><span class="ss">4.  </span>Formatting for rRNA filtering (these steps performed on computer cluster)</span>
<span id="cb14-82"><a href="#cb14-82"></a></span>
<span id="cb14-83"><a href="#cb14-83"></a>Get rRNAs from prokka which will be used later in the quality control steps to filter out reads mapping to bacterial rRNAs</span>
<span id="cb14-84"><a href="#cb14-84"></a></span>
<span id="cb14-85"><a href="#cb14-85"></a><span class="in">```</span></span>
<span id="cb14-86"><a href="#cb14-86"></a><span class="in">cd prokka</span></span>
<span id="cb14-87"><a href="#cb14-87"></a><span class="in">grep "ribosomal RNA" */*.tsv | sed 's/.tsv:/.ffn\t/g' | cut -f1,2 | while read FILE RNA; do seqkit grep -n -r -p "${RNA}" ${FILE}; done &gt; nanjingSynCom122_rrnas.fasta</span></span>
<span id="cb14-88"><a href="#cb14-88"></a><span class="in">seqkit rmdup -s nanjingSynCom122_rrnas.fasta</span></span>
<span id="cb14-89"><a href="#cb14-89"></a><span class="in">```</span></span>
<span id="cb14-90"><a href="#cb14-90"></a></span>
<span id="cb14-91"><a href="#cb14-91"></a><span class="fu">## Preparing for read mapping and feature counting</span></span>
<span id="cb14-92"><a href="#cb14-92"></a></span>
<span id="cb14-93"><a href="#cb14-93"></a><span class="ss">1. </span>Make SAF file from GFF file to be used with featureCounts</span>
<span id="cb14-94"><a href="#cb14-94"></a></span>
<span id="cb14-95"><a href="#cb14-95"></a><span class="in">```</span></span>
<span id="cb14-96"><a href="#cb14-96"></a><span class="in">cd rnaseq/map_count/saf</span></span>
<span id="cb14-97"><a href="#cb14-97"></a><span class="in">cat ../../../nanjingSynCom122/ids.txt | while read ID; do sed -n '/&gt;/q;p' ../../../nanjingSynCom122/prokka/${ID}/${ID}.gff | grep -v "^#" | sed 's/;.*$//g' | sed 's/ID=//g' | bioawk -t '{print $9, $1, $4, $5, $7}' | grep -v "^note"; done &gt; nanjingSynCom122_combined.saf</span></span>
<span id="cb14-98"><a href="#cb14-98"></a><span class="in">```</span></span>
<span id="cb14-99"><a href="#cb14-99"></a></span>
<span id="cb14-100"><a href="#cb14-100"></a>The resulting file looks like this</span>
<span id="cb14-101"><a href="#cb14-101"></a></span>
<span id="cb14-102"><a href="#cb14-102"></a><span class="in">```</span></span>
<span id="cb14-103"><a href="#cb14-103"></a><span class="in">head saf/nanjingSynCom122_combined.saf</span></span>
<span id="cb14-104"><a href="#cb14-104"></a><span class="in">```</span></span>
<span id="cb14-105"><a href="#cb14-105"></a></span>
<span id="cb14-106"><a href="#cb14-106"></a><span class="in">```         </span></span>
<span id="cb14-107"><a href="#cb14-107"></a><span class="in">GeneID  Chr Start   End Strand</span></span>
<span id="cb14-108"><a href="#cb14-108"></a><span class="in">1aci1_00001 1aci1   178 1365    +</span></span>
<span id="cb14-109"><a href="#cb14-109"></a><span class="in">1aci1_00002 1aci1   1498    2616    +</span></span>
<span id="cb14-110"><a href="#cb14-110"></a><span class="in">1aci1_00003 1aci1   2754    5378    +</span></span>
<span id="cb14-111"><a href="#cb14-111"></a><span class="in">1aci1_00004 1aci1   5603    5920    -</span></span>
<span id="cb14-112"><a href="#cb14-112"></a><span class="in">1aci1_00005 1aci1   6257    6868    -</span></span>
<span id="cb14-113"><a href="#cb14-113"></a><span class="in">1aci1_00006 1aci1   7122    7913    +</span></span>
<span id="cb14-114"><a href="#cb14-114"></a><span class="in">1aci1_00007 1aci1   7967    8389    +</span></span>
<span id="cb14-115"><a href="#cb14-115"></a><span class="in">1aci1_00008 1aci1   8463    8807    -</span></span>
<span id="cb14-116"><a href="#cb14-116"></a><span class="in">1aci1_00009 1aci1   8986    9246    +</span></span>
<span id="cb14-117"><a href="#cb14-117"></a><span class="in">```</span></span>
<span id="cb14-118"><a href="#cb14-118"></a></span>
<span id="cb14-119"><a href="#cb14-119"></a>Concatenate all assemblies into single file for mapping with bwa. Create the necessary indexes</span>
<span id="cb14-120"><a href="#cb14-120"></a></span>
<span id="cb14-121"><a href="#cb14-121"></a><span class="in">```</span></span>
<span id="cb14-122"><a href="#cb14-122"></a><span class="in">cd rnaseq/map_count/refs</span></span>
<span id="cb14-123"><a href="#cb14-123"></a><span class="in">cat ../../../nanjingSynCom122/prokka/*/*.fna &gt; nanjingSynCom122_combined.fna</span></span>
<span id="cb14-124"><a href="#cb14-124"></a><span class="in">samtools faidx nanjingSynCom122_combined.fna</span></span>
<span id="cb14-125"><a href="#cb14-125"></a><span class="in">bwa index nanjingSynCom122_combined.fna</span></span>
<span id="cb14-126"><a href="#cb14-126"></a><span class="in">```</span></span>
<span id="cb14-127"><a href="#cb14-127"></a></span>
<span id="cb14-128"><a href="#cb14-128"></a><span class="fu">## Quality control RNA-seq data</span></span>
<span id="cb14-129"><a href="#cb14-129"></a></span>
<span id="cb14-130"><a href="#cb14-130"></a>Here we do standard filtering and trimming of reads.Bacterial community total cDNA read pairs are processed using BBDuk (version 39.23</span>
<span id="cb14-131"><a href="#cb14-131"></a>(https://sourceforge.net/projects/bbmap/) to remove contaminants, trim reads that contained adapter sequence, right quality trim reads where quality drops below Q10, and exclude reads with more than 2 Ns. BBMap (version 39.23) was then used map cDNA read pairs to a database of bactieral 16S, 5S and 18S rRNAs from the 122 bacterial species SynCom based on exact matches.</span>
<span id="cb14-132"><a href="#cb14-132"></a></span>
<span id="cb14-133"><a href="#cb14-133"></a>The steps to reproduce this are available in <span class="in">`sh/qualitycontrol/`</span></span>
<span id="cb14-134"><a href="#cb14-134"></a></span>
<span id="cb14-135"><a href="#cb14-135"></a><span class="fu">## Map RNA-seq data</span></span>
<span id="cb14-136"><a href="#cb14-136"></a></span>
<span id="cb14-137"><a href="#cb14-137"></a>Here we map the quality controlled read pairs to a concatenated reference of the 122 bacterial SynCom genomes using BWA. The resulting BAM files are sorted and then the program featureCounts v2.1.1 is used to count the number of reads mapping to genes in the assemblies.</span>
<span id="cb14-138"><a href="#cb14-138"></a></span>
<span id="cb14-139"><a href="#cb14-139"></a>The steps to reproduce this are available in <span class="in">`sh/map_count/`</span></span>
<span id="cb14-140"><a href="#cb14-140"></a></span>
<span id="cb14-141"><a href="#cb14-141"></a><span class="fu">## Process data for differential expression</span></span>
<span id="cb14-142"><a href="#cb14-142"></a></span>
<span id="cb14-143"><a href="#cb14-143"></a><span class="ss">6.  </span>Load big matrix into R and use gene identifiers to partition into n <span class="sc">\&lt;</span>= 122 species-specific sub-matrices</span>
<span id="cb14-144"><a href="#cb14-144"></a></span>
<span id="cb14-145"><a href="#cb14-145"></a><span class="fu">## Compare Samples via Ordination</span></span>
<span id="cb14-146"><a href="#cb14-146"></a></span>
<span id="cb14-147"><a href="#cb14-147"></a>https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02337-8</span>
<span id="cb14-148"><a href="#cb14-148"></a></span>
<span id="cb14-149"><a href="#cb14-149"></a><span class="ss">7.  </span>Calculate TMM for each species sub-matrix separately.</span>
<span id="cb14-150"><a href="#cb14-150"></a></span>
<span id="cb14-151"><a href="#cb14-151"></a><span class="ss">-   </span>https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25</span>
<span id="cb14-152"><a href="#cb14-152"></a><span class="ss">-   </span>https://www.biostars.org/p/9475236/</span>
<span id="cb14-153"><a href="#cb14-153"></a><span class="ss">-   </span>https://www.biostars.org/p/306010/</span>
<span id="cb14-154"><a href="#cb14-154"></a><span class="ss">-   </span>https://www.biostars.org/p/419018/#419177</span>
<span id="cb14-155"><a href="#cb14-155"></a><span class="ss">-   </span>https://davetang.github.io/muse/edger.html</span>
<span id="cb14-156"><a href="#cb14-156"></a></span>
<span id="cb14-157"><a href="#cb14-157"></a><span class="ss">8.  </span>Combine TMM matrices for all species into a single matrix and use in a clustering analysis like tSNE (reproduces Fig. 3d)</span>
<span id="cb14-158"><a href="#cb14-158"></a></span>
<span id="cb14-159"><a href="#cb14-159"></a><span class="at">&gt; For the comparison of transcript abundance across multiple samples, transcript counts need to be normalized in each organism by individual transcript lengths and total library size.</span></span>
<span id="cb14-160"><a href="#cb14-160"></a><span class="at">&gt; Due to the differences in transcript abundances between the major and minor organisms, counts for each organism should be normalized independently using transcript per million (TPM) calculations </span><span class="sc">\[</span><span class="at">73</span><span class="sc">\]</span><span class="at">.</span></span>
<span id="cb14-161"><a href="#cb14-161"></a><span class="at">&gt; TPM values are calculated by dividing all read counts by the length of each gene in kilobases to obtain a reads per kilobase (RPK) value for each gene </span><span class="sc">\[</span><span class="at">92</span><span class="sc">\]</span><span class="at">.</span></span>
<span id="cb14-162"><a href="#cb14-162"></a><span class="at">&gt; The RPK value for each gene is then divided by the sum of RPK values divided by 1,000,000.</span></span>
<span id="cb14-163"><a href="#cb14-163"></a><span class="at">&gt; While RPKM and FPKM calculations are also used for normalization, the sum of the RPKM and FPKM values differ between samples with differing numbers of reads, which can result in disproportionate comparisons </span><span class="sc">\[</span><span class="at">92</span><span class="sc">\]</span><span class="at">.</span></span>
<span id="cb14-164"><a href="#cb14-164"></a></span>
<span id="cb14-165"><a href="#cb14-165"></a><span class="fu">## Find differentially expressed genes</span></span>
<span id="cb14-166"><a href="#cb14-166"></a></span>
<span id="cb14-167"><a href="#cb14-167"></a><span class="ss">9.  </span>Run a separate edgeR analysis for each species sub-matrix</span>
<span id="cb14-168"><a href="#cb14-168"></a><span class="ss">10. </span>Concatenate results of significantly differentially expressed genes</span>
<span id="cb14-169"><a href="#cb14-169"></a><span class="ss">11. </span>Send to collaborators</span>
<span id="cb14-170"><a href="#cb14-170"></a></span>
<span id="cb14-171"><a href="#cb14-171"></a><span class="fu">## Availability</span></span>
<span id="cb14-172"><a href="#cb14-172"></a></span>
<span id="cb14-173"><a href="#cb14-173"></a>Data and code in this GitHub repository (<span class="ot">&lt;https://github.com/GHUSERNAME/REPOSITORYNAME&gt;</span>) are provided under <span class="co">[</span><span class="ot">GNU AGPL3</span><span class="co">](https://www.gnu.org/licenses/agpl-3.0.html)</span>.</span>
<span id="cb14-174"><a href="#cb14-174"></a>The rendered project site is available at <span class="ot">&lt;https://GHUSERNAME.github.io/REPOSITORYNAME/&gt;</span>, which has been produced using <span class="co">[</span><span class="ot">Quarto notebooks</span><span class="co">](https://quarto.org/)</span>.</span>
<span id="cb14-175"><a href="#cb14-175"></a>The content on the rendered site is released under the <span class="co">[</span><span class="ot">CC BY 4.0.</span><span class="co">](https://creativecommons.org/licenses/by/4.0/)</span></span>
<span id="cb14-176"><a href="#cb14-176"></a>This repository hosts all code and data for this project, including the code necessary to fully recreate the rendered webpage.</span>
<span id="cb14-177"><a href="#cb14-177"></a></span>
<span id="cb14-178"><a href="#cb14-178"></a>An archived release of the code is available from Zenodo: <span class="ot">&lt;https://zenodo.org/records/EVENTUAL_ZENODO_RECORD&gt;</span></span>
<span id="cb14-179"><a href="#cb14-179"></a></span>
<span id="cb14-180"><a href="#cb14-180"></a>Raw sequencing data used in the project is available from NCBI Bioproject <span class="sc">\[</span>PRJNA00000000<span class="sc">\]</span>(https://www.ncbi.nlm.nih</span>
<span id="cb14-181"><a href="#cb14-181"></a></span>
<span id="cb14-182"><a href="#cb14-182"></a><span class="fu">## Reproducibility</span></span>
<span id="cb14-183"><a href="#cb14-183"></a></span>
<span id="cb14-184"><a href="#cb14-184"></a>The project uses <span class="co">[</span><span class="ot">`renv`</span><span class="co">](https://rstudio.github.io/renv/index.html)</span> to create a reproducible environment to execute the code in this project.</span>
<span id="cb14-185"><a href="#cb14-185"></a><span class="co">[</span><span class="ot">See here</span><span class="co">](https://rstudio.github.io/renv/articles/renv.html#collaboration)</span> for a brief overview on collaboration and reproduction of the entire project.</span>
<span id="cb14-186"><a href="#cb14-186"></a>To get up and running from an established repository, you could do:</span>
<span id="cb14-187"><a href="#cb14-187"></a></span>
<span id="cb14-188"><a href="#cb14-188"></a><span class="in">``` r</span></span>
<span id="cb14-189"><a href="#cb14-189"></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb14-190"><a href="#cb14-190"></a>renv<span class="sc">::</span><span class="fu">restore</span>()</span>
<span id="cb14-191"><a href="#cb14-191"></a><span class="in">```</span></span>
<span id="cb14-192"><a href="#cb14-192"></a></span>
<span id="cb14-193"><a href="#cb14-193"></a>To initiate <span class="in">`renv`</span> for a new project:</span>
<span id="cb14-194"><a href="#cb14-194"></a></span>
<span id="cb14-195"><a href="#cb14-195"></a><span class="in">``` r</span></span>
<span id="cb14-196"><a href="#cb14-196"></a><span class="fu">install.packages</span>(<span class="st">"renv"</span>)</span>
<span id="cb14-197"><a href="#cb14-197"></a><span class="co"># initialize</span></span>
<span id="cb14-198"><a href="#cb14-198"></a>renv<span class="sc">::</span><span class="fu">init</span>()</span>
<span id="cb14-199"><a href="#cb14-199"></a><span class="co"># install some new packages</span></span>
<span id="cb14-200"><a href="#cb14-200"></a>renv<span class="sc">::</span><span class="fu">install</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb14-201"><a href="#cb14-201"></a><span class="co"># record those packages in the lockfile</span></span>
<span id="cb14-202"><a href="#cb14-202"></a>renv<span class="sc">::</span><span class="fu">snapshot</span>()</span>
<span id="cb14-203"><a href="#cb14-203"></a><span class="in">```</span></span>
<span id="cb14-204"><a href="#cb14-204"></a></span>
<span id="cb14-207"><a href="#cb14-207"></a><span class="in">```{r}</span></span>
<span id="cb14-208"><a href="#cb14-208"></a><span class="co">#| eval: true</span></span>
<span id="cb14-209"><a href="#cb14-209"></a><span class="co">#| echo: false</span></span>
<span id="cb14-210"><a href="#cb14-210"></a><span class="co">#| output: false</span></span>
<span id="cb14-211"><a href="#cb14-211"></a><span class="co">#| error: false</span></span>
<span id="cb14-212"><a href="#cb14-212"></a><span class="co">#| warning: false</span></span>
<span id="cb14-213"><a href="#cb14-213"></a><span class="co"># If you want to automoate setting up some general directory structure you can do that here in a quarto chunk</span></span>
<span id="cb14-214"><a href="#cb14-214"></a><span class="co"># that does not get output to the final HTML. </span></span>
<span id="cb14-215"><a href="#cb14-215"></a></span>
<span id="cb14-216"><a href="#cb14-216"></a><span class="co"># Setup project structure</span></span>
<span id="cb14-217"><a href="#cb14-217"></a><span class="fu">library</span>(here)</span>
<span id="cb14-218"><a href="#cb14-218"></a><span class="fu">library</span>(fs)</span>
<span id="cb14-219"><a href="#cb14-219"></a></span>
<span id="cb14-220"><a href="#cb14-220"></a><span class="co"># define directory paths and overall structure</span></span>
<span id="cb14-221"><a href="#cb14-221"></a><span class="co"># the _data_raw directory is never touched or modified! </span></span>
<span id="cb14-222"><a href="#cb14-222"></a><span class="co"># All downstream work and intermediate results should be put in the data directory</span></span>
<span id="cb14-223"><a href="#cb14-223"></a></span>
<span id="cb14-224"><a href="#cb14-224"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"X"</span>, <span class="st">"Y01"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-225"><a href="#cb14-225"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"X"</span>, <span class="st">"Y02"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-226"><a href="#cb14-226"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"X"</span>, <span class="st">"Y03"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-227"><a href="#cb14-227"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"X"</span>, <span class="st">"Y04"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-228"><a href="#cb14-228"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"A"</span>, <span class="st">"B01"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-229"><a href="#cb14-229"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"A"</span>, <span class="st">"B02"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-230"><a href="#cb14-230"></a></span>
<span id="cb14-231"><a href="#cb14-231"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"X"</span>, <span class="st">"Y01"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-232"><a href="#cb14-232"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"X"</span>, <span class="st">"Y02"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-233"><a href="#cb14-233"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"X"</span>, <span class="st">"Y03"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-234"><a href="#cb14-234"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"X"</span>, <span class="st">"Y04"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-235"><a href="#cb14-235"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"A"</span>, <span class="st">"B01"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-236"><a href="#cb14-236"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"A"</span>, <span class="st">"B02"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-237"><a href="#cb14-237"></a></span>
<span id="cb14-238"><a href="#cb14-238"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>), <span class="at">recurse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-239"><a href="#cb14-239"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>